src/include/OpenImageIO/hash.h

```diff
@@ -81,6 +84,82 @@ inline size_t xxhash (const Str& s, size_t seed=1771) {
     return xxhash (s.data(), s.length(), seed);
 }
 
+template <typename T>
+class OIIO_API Hash {
+    enum { kSeed = 1738 + sizeof(T) };
+    void* m_state;
+    bool m_failed;
+public:
+    typedef T value_type;
+
+    Hash (T seed = kSeed);
+    ~Hash ();
+
+    void reset (T seed);
+    void append (const void* val, size_t N);
+    T value () const;
+
+    /// Hash known string objects
+    void append (const ustring &str);
+    void append (const string_view &str);
```

Comment:
I think append(string_view) is the only string one you need. Both std::string and char* will properly convert to string_view, that's string_view's whole point.

Also note that you want to pass string_view by value, not by reference. There's no advantage to passing string_view by reference (`string_view&`), and it will just incur the extra instructions of a double indirection.